.travis.yml

```diff
@@ -9,24 +9,78 @@ branches:
 
 matrix:
   include:
-    - python: 3.7
+    - os: linux
+      python: 3.7
       env: NOX_SESSION=check
-    - python: 3.7
+    - os: linux
+      python: 3.7
       env: NOX_SESSION=docs
 
-    - python: 3.6
+    - os: linux
+      python: 3.6
       env: NOX_SESSION=test-3.6
-    - python: 3.7
+    - os: linux
+      python: 3.7
       env: NOX_SESSION=test-3.7
-    - python: 3.8-dev
+
+    - os: linux
+      python: 3.8-dev
       env: NOX_SESSION=test-3.8
       dist: bionic # Required to get OpenSSL 1.1.1+
 
+    - os: windows
+      language: shell
+      python: 3.7
+      env:
+        PATH=/c/Python37:/c/Python37/Scripts:$PATH
+        NOX_SESSION=check
+      before_install:
+        - choco install python --version 3.7
+        - python -m pip install --upgrade pip
+    - os: windows
+      language: shell
+      python: 3.7
+      env:
+        PATH=/c/Python37:/c/Python37/Scripts:$PATH
+        NOX_SESSION=docs
+      before_install:
+        - choco install python --version 3.7
+        - python -m pip install --upgrade pip
+
+    - os: windows
+      language: shell
+      python: 3.6
+      env:
+        PATH=/c/Python36:/c/Python36/Scripts:$PATH
+        NOX_SESSION=test-3.6
+      before_install:
+        - choco install python --version 3.6
+        - python -m pip install --upgrade pip
+    - os: windows
+      language: shell
+      python: 3.7
+      env:
+        PATH=/c/Python37:/c/Python37/Scripts:$PATH
+        NOX_SESSION=test-3.7
+      before_install:
+        - choco install python --version 3.7
+        - python -m pip install --upgrade pip
+
+    - os: windows
+      language: shell
+      python: 3.8-dev
+      env:
+        PATH=/c/Python38:/c/Python38/Scripts:$PATH
+        NOX_SESSION=test-3.8
+      before_install:
+        - choco install python --pre
+        - python -m pip install --upgrade pip
   # Allow builds to fail when using '-dev' Python versions
   # Should still investigate failures periodically for new APIs.
```

Comment:
Should we move this comment so that it's clearer that it targets `- python: 3.8-dev`?

```yaml
fast_finish: true
allow_failures:
  - os: windows
  # Allow builds to fail when using '-dev' Python versions.
  # We should still investigate failures periodically for new APIs.
  - python: 3.8-dev
```